### PR TITLE
feat: icac

### LINKED ISSUE #8245: 香港廉政公署
### 网站地址
https://www.icac.org.hk/tc/home/index.html

### 网站描述
廉政公署，簡稱「廉署」，於1974年2月15日根據《總督特派廉政專員公署條例》成立，當時名叫總督特派廉政專員公署，是獨立及直接向香港政府首長負責的紀律部隊及執法機構，以肅貪倡廉為目標，一直以來透過執法、預防及教育「三管齊下」的策略打擊貪污。-wiki

### 需要生成什么内容？
新聞公佈 https://www.icac.org.hk/tc/press/index.html

### PR DESCRIPTION
**Pull request by linbuxiao**
## 该 PR 相关 Issue / Involved issue

Close #8245

## 完整路由地址 / Example for the proposed route(s)

```routes
/icac/news/:lang?
```

## 新RSS检查列表 / New RSS Script Checklist
  
- [x] New Route
- [x] Documentation
  - [x] CN
  - [x] EN
- [x] 全文获取 fulltext
  - [x] Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] 日期和时间 date and time
  - [x] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added 
- [ ] `Puppeteer`

## 说明 / Note
新增香港廉政公署新闻公布

### CODE REVIEW

**Comment by linbuxiao** [comment]
@NeverBehave 新路由提交，麻烦大佬review~

File: docs/government.md [hunk 1]
```diff
@@ -707,3 +707,9 @@ pageClass: routes
 ### 审查调查
 
 <Route author="LogicJake" example="/ccdi/scdc" path="/ccdi/scdc"/>
+
+## 香港廉政公署
+
+### 新闻公布
+
+<Route author="linbuxiao" example="/ccdi/news" path="/ccdi/news"/>
```
Comment (TonyRL) [inline docs/government.md:715]:
```suggestion
<Route author="linbuxiao" example="/icac/news" path="/icac/news"/>
```

**Review by TonyRL — CHANGES_REQUESTED** [review]

File: lib/v2/icac/maintainer.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+    news: ['linbuxiao'],
```
Comment (TonyRL) [inline lib/v2/icac/maintainer.js:2]:
```suggestion
    '/news': ['linbuxiao'],
```

**Review by TonyRL — CHANGES_REQUESTED** [review]

File: lib/v2/icac/radar.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+    'https://www.icac.org.hk/tc/press/index.html': {
+        _name: '廉政公署',
+        '.': [
+            {
+                title: '新闻公布',
+                docs: 'https://docs.rsshub.app/government.html#xiang-gang-lian-zheng-gong-shu',
+                source: ['/', '/'],
```
Comment (TonyRL) [inline lib/v2/icac/radar.js:8]:
```suggestion
                source: ['/'],
```

**Review by TonyRL — CHANGES_REQUESTED** [review]

File: docs/government.md (hunk 1 above)
Comment (linbuxiao) [reply docs/government.md:715]: Thank you! I guess I was too sloppy.

File: lib/v2/icac/radar.js (hunk 3 above)
Comment (linbuxiao) [reply lib/v2/icac/radar.js:8]: 谢谢，之前对`source`的意义不太明了。

**Comment by TonyRL** [comment]
I think it’s a nice to if you could add support for `sc` and `en` language. They share almost the same url.
`https://www.icac.org.hk/sc/press/index.html`
`https://www.icac.org.hk/en/press/index.html`

**Comment by linbuxiao** [comment]
> I think it’s a nice-to-have if you could add support for `sc` and `en` language. They share almost the same url.
> `https://www.icac.org.hk/sc/press/index.html`
> `https://www.icac.org.hk/en/press/index.html`

This is a very good idea! I will submit it right away.

File: docs/en/government.md [hunk 4]
```diff
@@ -67,3 +67,9 @@ Category
 ### Dispute settlement news
 
 <RouteEn author="nczitzk" example="/wto/dispute-settlement" path="/wto/dispute-settlement/:year?" :paramsDesc="['Year, current year by default']"/>
+
+## Hong Kong Independent Commission Against Corruption
+
+### Press Releases
+
+<Route author="linbuxiao" example="/icac/news/sc" path="/icac/news/:lang?" :paramsDesc="['Language, current year by sc. Supprot sc and en']">
```
Comment (TonyRL) [inline docs/en/government.md:75]:
```suggestion
<Route author="linbuxiao" example="/icac/news/sc" path="/icac/news/:lang?" :paramsDesc="['Language, default to `sc`. Supprot `sc`, `tc` and `en`']">
```

File: docs/government.md [hunk 5]
```diff
@@ -707,3 +707,9 @@ pageClass: routes
 ### 审查调查
 
 <Route author="LogicJake" example="/ccdi/scdc" path="/ccdi/scdc"/>
+
+## 香港廉政公署
+
+### 新闻公布
+
+<Route author="linbuxiao" example="/icac/news/sc" path="/icac/news/:lang?" :paramsDesc="['语言, 留空为sc, 支持sc（中文）, en（英文）']">
```
Comment (TonyRL) [inline docs/government.md:715]:
```suggestion
<Route author="linbuxiao" example="/icac/news/sc" path="/icac/news/:lang?" :paramsDesc="['语言，留空为`sc`，支持`sc`（简中），`tc`（繁中），en（英文）']">
```

File: lib/v2/icac/news.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+                item.description = des + thumbs;
+                return item;
+            })
+        )
+    );
+    ctx.state.data = {
+        title: 'ICAC 新闻公布',
+        link: 'https://www.icac.org.hk/tc/press/index.html',
+        description: 'ICAC 新闻公布',
+        language: ctx.params.lang || 'sc',
```
Comment (TonyRL) [inline lib/v2/icac/news.js:44]: ISO 639-1: `zh-cn`, `zh-hk`, `en-us`

**Review by TonyRL — CHANGES_REQUESTED** [review]

File: lib/v2/icac/news.js (hunk 6 above)
Comment (linbuxiao) [reply lib/v2/icac/news.js:44]: `current year by sc`
Hahahahaha I copied it and forgot to change it. I will format it soon. Thank you.

File: docs/government.md [hunk 7]
```diff
@@ -707,3 +707,9 @@ pageClass: routes
 ### 审查调查
 
 <Route author="LogicJake" example="/ccdi/scdc" path="/ccdi/scdc"/>
+
+## 香港廉政公署
+
+### 新闻公布
+
+<Route author="linbuxiao" example="/icac/news/sc" path="/icac/news/:lang?" :paramsDesc="['语言，留空为`sc`，支持`sc`（简中），`tc`（繁中），en（英文）']">
```
Comment (TonyRL) [inline docs/government.md:715]:
```suggestion
<Route author="linbuxiao" example="/icac/news/sc" path="/icac/news/:lang?" :paramsDesc="['语言，留空为`sc`，支持`sc`（简中），`tc`（繁中），`en`（英文）']">
```

File: lib/v2/icac/news.js (hunk 6 above)
Comment (linbuxiao) [reply lib/v2/icac/news.js]: Add it at f9188fd.

**Comment by linbuxiao** [comment]
prettier's semicolon fixes seem to fail sometimes, so that I have a few more commits to fix them manually.

**Comment by linbuxiao** [comment]
@TonyRL Have added your id to the router maintainer, thank you for your guidance.

File: lib/v2/icac/radar.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+    'https://www.icac.org.hk': {
+        _name: '廉政公署',
+        '.': [
+            {
+                title: '新闻公布',
+                docs: 'https://docs.rsshub.app/government.html#xiang-gang-lian-zheng-gong-shu',
+                source: ['/:lang/press/index.html'],
+                target: '/news/:lang',
```
Comment (TonyRL) [inline lib/v2/icac/radar.js:9]:
```suggestion
                target: '/icac/news/:lang',
```
[script-standard](https://docs.rsshub.app/joinus/script-standard.html) has a typo in radar rules.
[quick-start](https://docs.rsshub.app/joinus/quick-start.html#ti-jiao-xin-de-rsshub-radar-gui-ze) has the right format.

**Review by TonyRL — CHANGES_REQUESTED** [review]

File: lib/v2/icac/radar.js (hunk 8 above)
Comment (linbuxiao) [reply lib/v2/icac/radar.js:9]:
```javascript
module.exports = {
    'furstar.jp': {
        _name: 'Furstar',
        '.': [
            {
                title: '最新售卖角色列表',
                docs: 'https://docs.rsshub.app/shopping.html#furstar-zui-xin-shou-mai-jiao-se-lie-biao',
                source: ['/:lang', '/'],
                target: '/characters/:lang',
            },
            {
                title: '已经出售的角色列表',
                docs: 'https://docs.rsshub.app/shopping.html#furstar-yi-jing-chu-shou-de-jiao-se-lie-biao',
                source: ['/:lang/archive.php', '/archive.php'],
                target: '/archive/:lang',
            },
            {
                title: '画师列表',
                docs: 'https://docs.rsshub.app/shopping.html#furstar-hua-shi-lie-biao',
                source: ['/'],
                target: '/artists',
            },
        ],
    },
};
```

In the example I noticed that the route root path was not added to target. Should I add it here? Thanks.

Comment (linbuxiao) [reply lib/v2/icac/radar.js:9]: > ```js
> module.exports = {
>     'furstar.jp': {
>         _name: 'Furstar',
>         '.': [
>             {
>                 title: '最新售卖角色列表',
>                 docs: 'https://docs.rsshub.app/shopping.html#furstar-zui-xin-shou-mai-jiao-se-lie-biao',
>                 source: ['/:lang', '/'],
>                 target: '/characters/:lang',
>             },
>             {
>                 title: '已经出售的角色列表',
>                 docs: 'https://docs.rsshub.app/shopping.html#furstar-yi-jing-chu-shou-de-jiao-se-lie-biao',
>                 source: ['/:lang/archive.php', '/archive.php'],
>                 target: '/archive/:lang',
>             },
>             {
>                 title: '画师列表',
>                 docs: 'https://docs.rsshub.app/shopping.html#furstar-hua-shi-lie-biao',
>                 source: ['/'],
>                 target: '/artists',
>             },
>         ],
>     },
> };
> ```
> 
> In the example I noticed that the `route root path` was not added to `target`. Should I add it here? Thanks.

Oh! I read it backwards, I thought the other one was wrong. Thanks!

**Review by NeverBehave — APPROVED** [review]

**Comment by NeverBehave** [comment]
🎉

**Comment by linbuxiao** [comment]
> 🎉

thank you!

File: docs/government.md [hunk 9]
```diff
@@ -707,3 +707,9 @@ pageClass: routes
 ### 审查调查
 
 <Route author="LogicJake" example="/ccdi/scdc" path="/ccdi/scdc"/>
+
+## 香港廉政公署
+
+### 新闻公布
+
+<Route author="linbuxiao" example="/icac/news/sc" path="/icac/news/:lang?" :paramsDesc="['语言，留空为`sc`，支持`sc`（简中），`tc`（繁中），`en`（英文）']">
```
Comment (TonyRL) [inline docs/government.md:715]:
```suggestion
<Route author="linbuxiao" example="/icac/news/sc" path="/icac/news/:lang?" :paramsDesc="['语言，留空为`sc`，支持`sc`（简中），`tc`（繁中），`en`（英文）']"/>
```

File: docs/en/government.md [hunk 10]
```diff
@@ -67,3 +67,9 @@ Category
 ### Dispute settlement news
 
 <RouteEn author="nczitzk" example="/wto/dispute-settlement" path="/wto/dispute-settlement/:year?" :paramsDesc="['Year, current year by default']"/>
+
+## Hong Kong Independent Commission Against Corruption
+
+### Press Releases
+
+<Route author="linbuxiao" example="/icac/news/sc" path="/icac/news/:lang?" :paramsDesc="['Language, default to `sc`. Supprot `en`(English), `sc`(Simplified Chinese) and `tc`(Traditional Chinese)']">
```
Comment (TonyRL) [inline docs/en/government.md:75]:
```suggestion
<RouteEn author="linbuxiao" example="/icac/news/sc" path="/icac/news/:lang?" :paramsDesc="['Language, default to `sc`. Supprot `en`(English), `sc`(Simplified Chinese) and `tc`(Traditional Chinese)']"/>
```

Comment (TonyRL) [reply docs/en/government.md:75]: This should fix the issue of route not searchable in https://docs.rsshub.app

File: docs/government.md (hunk 9 above)
Comment (TonyRL) [reply docs/government.md:715]: This should fix the issue of route not searchable in https://docs.rsshub.app

File: docs/en/government.md (hunk 10 above)
Comment (linbuxiao) [reply docs/en/government.md:75]: I will make changes to this in the next PR~

Comment (linbuxiao) [reply docs/en/government.md:75]: pr #8273

**Comment by wft44maqb** [comment]
`Error message: wrong path` @linbuxiao @NeverBehave

**Comment by NeverBehave** [comment]
@wft44maqb https://github.com/DIYgod/RSSHub/issues/8254

还有，这种明显相同的问题，不要反复发，邮件提醒很烦的

**Pull request merged**